features/admin.administrators.v1/wizard/steps/admin-user-basic.tsx

```diff
@@ -47,99 +44,140 @@ import {
-    triggerSubmit: boolean;
     onSubmit: (values: any) => void;
     administratorType: string;
     setFinishButtonDisabled: (values: boolean) => void;
+    /**
+     * Roles list for console application.
+     */
+    consoleRolesList: RolesInterface[];
+    /**
+     * Whether the console roles list is loading.
+     */
+    isConsoleRolesListLoading: boolean;
+}
+
+export interface AddAdminUserBasicFormRef {
+    triggerSubmit: () => void;
 }
 
 /**
  * Add admin user basic component.
  *
  * @returns add admin modal component.
  */
-export const AddAdminUserBasic: React.FunctionComponent<AddAdminUserBasicProps> = (
-    props: AddAdminUserBasicProps): ReactElement => {
+export const AddAdminUserBasic: ForwardRefExoticComponent<RefAttributes<AddAdminUserBasicFormRef> &
+AddAdminUserBasicProps> = forwardRef((
+    props: AddAdminUserBasicProps,
+    ref: ForwardedRef<AddAdminUserBasicFormRef>
+): ReactElement => {
 
     const {
         administratorType,
-        triggerSubmit,
         onSubmit,
         setFinishButtonDisabled,
-        [ "data-componentid"]: componentId
+        consoleRolesList,
+        isConsoleRolesListLoading,
+        [ "data-componentid"]: componentId = "add-admin-user-basic"
     } = props;
 
-    const [ userRoleOptions, setUserRoleList ] = useState([]);
-    const [ rolesList, setRolesList ] = useState<RolesInterface[]>([]);
     const [ usersList, setUsersList ] = useState<UserBasicInterface[]>([]);
     const [ checkedAssignedListItems, setCheckedAssignedListItems ] = useState<UserBasicInterface[]>([]);
     const [ isUserListRequestLoading, setUserListRequestLoading ] = useState<boolean>(false);
-    const [ isUserRoleOptionsRequestLoading, setUserRoleOptionsRequestLoading ] = useState<boolean>(false);
-    const [ listItemLimit, setListItemLimit ] = useState<number>(0);
     const [ userListMetaContent, setUserListMetaContent ] = useState(undefined);
     const [ searchQuery, setSearchQuery ] = useState<string>("");
     const [ listOffset ] = useState<number>(0);
 
     const { t } = useTranslation();
     const { getLink } = useDocumentation();
 
+    const triggerFormSubmit: MutableRefObject<() => void> = useRef<(() => void) | null>(null);
+
     const consoleSettingsFeatureConfig: FeatureAccessConfigInterface =
         useSelector((state: AppState) => state?.config?.ui?.features?.consoleSettings);
     const isInvitedAdminInConsoleSettingsEnabled: boolean = !consoleSettingsFeatureConfig?.disabledFeatures?.includes(
         "consoleSettings.invitedExternalAdmins"
     );
 
-    /**
-     * Retrieve the application data for the console application, filtering by name.
-     */
-    const { data: applicationListData } = useApplicationList(
-        null,
-        null,
-        null,
-        `name eq ${ApplicationManagementConstants.CONSOLE_APP_NAME}`
-    );
+    // Username input validation error messages.
+    const USERNAME_REGEX_VIOLATION_ERROR_MESSAGE: string = t("users:guestUsers.fields." +
```

Comment:
Is it worth to maintain this in a const?

Reply:
This was already there in the file. Hence didn't change it
https://github.com/wso2/identity-apps/pull/7754/files#diff-17c4f3a77525b8e5694eff81ac171175a8938562ccda99ab6381c6dc4d7680aeL130